ocaml/typing/includecore.ml

```diff
@@ -554,6 +555,10 @@ let report_type_mismatch first second decl env ppf err =
       pr "Their internal representations differ:@ %s %s %s."
          (choose ord first second) decl
          "is extensible"
+  | With_null_representation ord ->
+      pr "Their internal representations differ:@ %s %s %s."
```

Comment:
This can't be tested yet, because `Variant_with_null` is not completely implemented and so we can't set `or_null`'s kind to it. This will be tested in the final PR.